### DISCUSSION TITLE
Super Micro X10DRU-i+ zones

### DISCUSSION THREAD

**Discussion by petersulyok** [petersulyok/smfc#48] [General]
Hi,

I have an idea that this motherboard has not only two but multiple zones.
You may play with this command

`# ipmitool raw 0x30 0x70 0x66 0x01 zone 50`

substituting a value for zone parameter (e.g. 0, 1, 2, 3, 4) ad report back the output of the command execution.
This command will set the fan level to 50% in the specified zone. With changing the zone parameter your can discover the potential zones on your motherboard. Of course, connected fans may make this experience easier.

**Comment by petersulyok** [comment]
Hi @alexhelms, any feedback on this?

**Comment by MrMeeb** [comment]
Hi Peter, I also have an X10DRU-i+ in a SYS-6028U-TR4T+ - it has 4 internal chassis fans. When enabling SMFC in it's default state, the fans respond in one half of the server but not the other it would seem: 

```
FAN1             | 4800.000   | RPM        | ok    | 300.000   | 500.000   | 700.000   | 25300.000 | 25400.000 | 25500.000 
FAN2             | 4600.000   | RPM        | ok    | 300.000   | 500.000   | 700.000   | 25300.000 | 25400.000 | 25500.000 
FAN3             | na         |            | na    | na        | na        | na        | na        | na        | na        
FAN4             | na         |            | na    | na        | na        | na        | na        | na        | na        
FAN5             | na         |            | na    | na        | na        | na        | na        | na        | na        
FAN6             | na         |            | na    | na        | na        | na        | na        | na        | na        
FAN7             | 9300.000   | RPM        | ok    | 300.000   | 500.000   | 700.000   | 25300.000 | 25400.000 | 25500.000 
FAN8             | 9300.000   | RPM        | ok    | 300.000   | 500.000   | 700.000   | 25300.000 | 25400.000 | 25500.000 
```

I'm not all that familiar with `ipmitool`, but entering your suggested command of `ipmitool raw 0x30 0x70 0x66 0x01 zone 50` showed zones `0` controlling `FAN1` and `FAN2`, and `1` controlling `FAN7` and `FAN8`. 

It seems that SMFC is only controlling zone 0 currently - setting both zones to 75% with `ipmitool raw 0x30 0x70 0x66 0x01 zone 75`, then enabling SMFC via `systemctl start smfc.service` only sees `FAN1` and `FAN2` drop in speed.

Config, for good measure:

```#
#   smfc.conf
#   smfc service configuration parameters
#


[Ipmi]
# Path for ipmitool (str, default=/usr/bin/ipmitool)
command=/usr/bin/ipmitool 
# Delay time after changing IPMI fan mode (int, seconds, default=10)
fan_mode_delay=10
# Delay time after changing IPMI fan level (int, seconds, default=2)
fan_level_delay=2
# CPU and HD zones are swapped (bool, default=0).
swapped_zones=0


[CPU zone]
# Fan controller enabled (bool, default=0)
enabled=1
# Number of CPUs (int, default=1)
count=1
# Calculation method for CPU temperatures (int, [0-minimum, 1-average, 2-maximum], default=1)
temp_calc=1
# Discrete steps in mapping of temperatures to fan level (int, default=6)
steps=6
# Threshold in temperature change before the fan controller reacts (float, C, default=3.0)
sensitivity=3.0
# Polling time interval for reading temperature (int, sec, default=2)
polling=2
# Minimum CPU temperature (float, C, default=30.0)
min_temp=30.0
# Maximum CPU temperature (float, C, default=60.0)
max_temp=60.0
# Minimum CPU fan level (int, %, default=35)
min_level=35
# Maximum CPU fan level (int, %, default=100)
max_level=100
# Optional parameter, it will be generated automatically for Intel CPUs and must be specified manually for AMD CPUs.
# Path for CPU sys/hwmon file(s) (str multi-line list, default=/sys/devices/platform/coretemp.0/hwmon/hwmon*/temp1_input)
# hwmon_path=/sys/devices/platform/coretemp.0/hwmon/hwmon*/temp1_input
#            /sys/devices/platform/coretemp.1/hwmon/hwmon*/temp1_input
# or
# hwmon_path=/sys/bus/pci/drivers/k10temp/0000*/hwmon/hwmon*/temp1_input


[HD zone]
# Fan controller enabled (bool, default=0)
enabled=0
# Number of HDs (int, default=1)
count=1
# Calculation of HD temperatures (int, [0-minimum, 1-average, 2-maximum], default=1)
temp_calc=1
# Discrete steps in mapping of temperatures to fan level (int, default=4)
steps=4
# Threshold in temperature change before the fan controller reacts (float, C, default=2.0)
sensitivity=2.0
# Polling interval for reading temperature (int, sec, default=10)
polling=10
# Minimum HD temperature (float, C, default=32.0)
min_temp=32.0
# Maximum HD temperature (float, C, default=46.0)
max_temp=46.0
# Minimum HD fan level (int, %, default=35)
min_level=35
# Maximum HD fan level (int, %, default=100)
max_level=100
# Names of the HDs (str multi-line list, default=)
# These names MUST BE specified in '/dev/disk/by-id/...' form!
hd_names=
# Optional parameter, it will be generated automatically based on the disk names.
# List of files in /sys/hwmon file system or 'hddtemp' (str multi-line list, default=)
# hwmon_path=/sys/class/scsi_disk/0:0:0:0/device/hwmon/hwmon*/temp1_input
#            /sys/class/scsi_disk/1:0:0:0/device/hwmon/hwmon*/temp1_input
#            hddtemp
# Standby guard feature for RAID arrays (bool, default=0)
standby_guard_enabled=0
# Number of HDs already in STANDBY state before the full RAID array will be forced to it (int, default=1)
standby_hd_limit=1
# Path for 'smartctl' command (str, default=/usr/sbin/smartctl).
# Required for 'standby guard' feature only
smartctl_path=/usr/sbin/smartctl
# Path for 'hddtemp' command (str, default=/usr/sbin/hddtemp).
# Required for reading of the temperature of SAS/SCSI disks.
hddtemp_path=/usr/sbin/hddtemp```

**Comment by petersulyok** [comment]
Hi @MrMeeb, I'm happy to investigate more on this. My assumption was that this motherboard has multiple CPU zones defined in IPMI. Some further questions:
- How many CPUs are installed on this board?
- How many fans are installed on this board? 
- I assume you intentionally disabled HD zone in your configuration file, right?

Based on your report my assumption is that the CPU zone will be FAN1 and FAN2 and the HD zone will be FAN7 and FAN8.
What if you run `ipmitool` with higher zone values like 2, 3, 4?

**Comment by MrMeeb** [comment]
Hi Peter,

2 CPUs installed - it's a dual socket system. There are 4 fans installed in the 2U chassis, attached to (based on the output of `impitool sensor` fan headers 1, 2, 7 and 8. 

Running `ipmitool` on higher zones has no audible effect. Of course if it's affecting fans that aren't connected I have no way of telling, but the fact that zone 1 affects fans 7 and 8 suggests it split in half as you said.

I disabled HD zone, yeah. Sorry I forgot to mention that. I was having trouble with it not detecting drives, even SATA ones, via `drivetemp`. I know others in #21 said that that SATA drives on a SAS backplane seemed to work with `drivetemp` but that doesn't seem to be the case for me. I wanted to just test SMFC quickly, so didn't troubleshoot it. I have now resolved the issue by switching all drives to use `hddtemp`, even the SATA ones.

Having done that and now enabled HD zone, I can see that you're right - zone 0 for CPU and zone 1 for drives. However in a rack server I'm not sure if this makes total sense - all 4 fans are responsible for cooling all HDDs and both CPUs. I think having all fans react off CPU temp, unless HDD temp passes a certain threshold, could make sense, but even then, in a mixed load of SSDs and HDDs, what's considered a safe operating temperature varies greatly.

As a side note, is it necessary to set `hddtemp` for every drive added to the list, or is it possible to tell SMFC to use `hddtemp` for everything? If the numbers in  `hd_names` and `hwmon_path` don't match, an error is thrown. You mentioned `temp_source` in #21 but it doesn't seem to work - I guess you chose to not implement that option?

```
[HD zone]
# Fan controller enabled (bool, default=0)
enabled=1
# Number of HDs (int, default=1)
count=6
# Calculation of HD temperatures (int, [0-minimum, 1-average, 2-maximum], default=1)
temp_calc=1
# Discrete steps in mapping of temperatures to fan level (int, default=4)
steps=4
# Threshold in temperature change before the fan controller reacts (float, C, default=2.0)
sensitivity=2.0
# Polling interval for reading temperature (int, sec, default=10)
polling=10
# Minimum HD temperature (float, C, default=32.0)
min_temp=32.0
# Maximum HD temperature (float, C, default=46.0)
max_temp=46.0
# Minimum HD fan level (int, %, default=35)
min_level=35
# Maximum HD fan level (int, %, default=100)
max_level=100
# Names of the HDs (str multi-line list, default=)
# These names MUST BE specified in '/dev/disk/by-id/...' form!
hd_names=/dev/disk/by-id/ata-Samsung_SSD_860_EVO_500GB_S3Z2NB0K998710F
        /dev/disk/by-id/ata-Samsung_SSD_860_EVO_500GB_S3Z2NB0NA07832R
        /dev/disk/by-id/scsi-35000cca03e42178c
        /dev/disk/by-id/scsi-35000cca03e4efc48
        /dev/disk/by-id/scsi-35000cca03e50c834
        /dev/disk/by-id/scsi-35000cca04608e71c
# Optional parameter, it will be generated automatically based on the disk names.
# List of files in /sys/hwmon file system or 'hddtemp' (str multi-line list, default=)
hwmon_path=hddtemp
        hddtemp
        hddtemp
        hddtemp
        hddtemp
        hddtemp
#            /sys/class/scsi_disk/1:0:0:0/device/hwmon/hwmon*/temp1_input
#            hddtemp
# Standby guard feature for RAID arrays (bool, default=0)
standby_guard_enabled=0
# Number of HDs already in STANDBY state before the full RAID array will be forced to it (int, default=1)
standby_hd_limit=1
# Path for 'smartctl' command (str, default=/usr/sbin/smartctl).
# Required for 'standby guard' feature only
smartctl_path=/usr/sbin/smartctl
# Path for 'hddtemp' command (str, default=/usr/sbin/hddtemp).
# Required for reading of the temperature of SAS/SCSI disks.
hddtemp_path=/usr/sbin/hddtemp
```

**Comment by petersulyok** [comment]
Hi Charlie,

I'm thinking on how I could support this scenario and I've got two ideas in my mind:
1. You may try to put the four fans into a common zone. For example you may try to put fan7 and fan8 to other connector and in this case all four fans will be working in the same zone.
2. I'm thinking on the implementation of a new feature where `smfc` could support PC/Server cases with one cooling chamber where the fans are common but there multiple heat sources (HDs, CPUs). I need more time to figure it out the best way.

BTW, would be interesting to discover the other fan connectors on the motherboard. Would be interesting to understand the mapping between FANS and zones.

**Comment by MrMeeb** [comment]
Hi Peter,

The length of the fan cables internally mean I can connect all fans to zone 1 in headers 5-8. Good idea. I've done that and swapped the zones so CPU is using zone 1, and the server is currently just not reacting to storage temperature at all. I'd say that being able to have multiple heat sources in a common fan zone would be useful in the long run, but this is definitely workable. Also, every other server I've had/built doesn't account for storage temps anyway. It get's a bit complicated with the different operating temperatures of different storage techs, unless you can specify acceptable temp ranges per drive.

I have tested the other fan headers. Fans 1-4 belong to zone 0 and 5-8 belong to zone 1. I also realised I missed `FAN9`. I believe this is on one of the rear risers in the server, to then be plumbed to an external connector on the rear of the chassis. I don't currently have a fan hanging around to test that one unfortunately, and not looking to cut up cable-ties on chassis fans to see if a cable can reach.

**Comment by OdinVex** [comment]
I have the same server but I'm not sure I'd like putting the fans into one zone, I don't know if they're on one rail of power that might be overdrawing or not.

**Comment by duecedriver** [comment]
again this comes down to adding a bit of customization in the provisioning script

you could ask for a comma delineated list of of fans for each zone.. ie  zone 0 1,2,3,4  zone b 1,b

or in the case above .. zone 0 1,2 zone 1 7,8

its housekeeping really  as there are only 2 zones to control in the end.. 

its about attaching the correct sensor to each zone

doing it this way you could query the user which sensor for each fan logical fan group and let the user decide

in this example CPU0 zone 0 CPU1 zone 1

perhaps only using HDD temp (on the wall in front of the fans) as a high critical over ride.. to bump the fan curve for both zones during high drive activity.. but unless you are running data base ops.. I have never had the hard drives be the issue with any active cooling on them... zfs scub even .. again.. unless you are running chettas from the good ole days

**Comment by OdinVex** [comment]
> perhaps only using HDD temp

There aren't any FANA+, purely FAN0+ (edit: or did you mean use actual HDD temps in script, not from polling IPMI?). I'll just roll out a script. I have to account for MB_NIC_TEMP1 and MB_NIC_TEMP2 as well since they have no cooling (AOC-2UR6-i4XT on the other end of the motherboard, for example) and are done mostly by two of the fans from the far end just hoping to blow on them. I've been considering mounting a small 30x30x10mm fan directly onto their combined heat-sink and running a PWM cable to it off a spare port on a motherboard from one of the sections that usually has a lower fan speed, just to be safe about the power. I'm aware a tiny 3010 fan like that draws nearly nothing, I just don't willy-nilly draw from rails. Not as much air gets through if the PCIe slot is used, but especially if a rear-drive-kit is modded in (a rivet or two needed removal). Just trying to account for everything.

I wish I could describe the fan layout to the scripts such "these fans are behind these drives and affect this other sensor", that's why I'm thinking I'll roll my own given my needs.

**Comment by duecedriver** [comment]
the ideal solution In my eyes..

there should be a script either stand alone or part of the installer.. they both have pro-con
to ID the fans and the zones that are available as well as a list of sensors to be used as 'IDs' 
either during provisioning or before hand.

the script should, during provisioning 
>set up or permit the use of all available zones
> ID which fans go into which zone
> ID a 'PRIMARY' sensor for each zone to run PID loop on
> ID any other sensors in the zone that while not in the pid loop would trigger failsafe if their values hit their assigned values
> permit assigning those trigger values to all fans/sensors
> give the user the choice of failsafe and exit stock setting instead of MAX.. I suggest optimal as default as that is factory fan curve
> permit the user to assess min and max fan RPM to assist in fan curve, prime95 can be called to assess if the max fan is enough... or possibly use it to auto tune the fan curve

**Comment by petersulyok** [comment]
Hi @duecedriver and @OdinVex, this is IPMI, please think about zones instead of individual fans. You can decide which zone will contain the most of the fans and how your cooling will work. This could be tough for server cases.

In my mind an an ideal case in the config file:
- the IPMI zone id could be defined freely (e.g. 0, 1, 2, ....) instead of builtin 0 and 1.
- different temperature sources could be defined (e.g. CPUs, HDDs, GPUs, etc.) 
- a temperature source could be assigned to a zone

My problem is that the configuration of `smfc` is already very complex and not really user-friendly. These steps would make it even more complex and abstract.

**Comment by duecedriver** [comment]
I know this is IPMI .. and unfortunately.. super micros IPMI is horrible... hence the 2 zones.. other boards I have for example the intel S5520HC... far more configurable and fan curve support etc is built in.. 


this isn't rocket science .. and with super micro like you said.. you only have 2 zones to deal with.. the problem is that with supermicro .. they build mostly for enterprise and a lot to spec.. so a board on their site that appear 'retail' is likely all but bought by the thousands for data centers or integrators.. and they get to call the shots on customizations and changes to firmware.

what makes smfc poos is its LACK of transparency... ie customization.

as I stated.. in order for this to function well for more than just one board, with one layout is what I specified..

users need to be able to identify how THEIR particular board is layed out and which fan headers are in which zones.. probably impossible to figure out what the actual bmc fan curves are.. but at least you know what fans you are controlling.

then you need to be able to specify which IPMI sensor you wish to control each zone... to either over ride what BMC is doing.. you are doing it anyway.. that is why you need to select MAX for the zone MAX overrides the BMC fan curve and takes the BMC logic out of the equation..

You need to do this because each board is different .. 

in many of the server 2 processor designs .. its designed to have a zone in front of each heatsink, controlled by CPU temp.. so the peripheral zone is not controller on board thep its controlled by CPU1 and the other zone by CPU0.


The reason also that other than specifying just a primary sensor is this .. lets discuss this example where primary heuristics are controlled by CPU temp.. if you completely ignore every other temp in the case.. you will burn something up.

the way the BMC currently works.. is that if  ANY sensor goes above its set High Unrecoverable, it will set all fans/zones to max... even with your script running this MAY still happen, again, different in each setup depending on the underlying BMC 

so as temps in the HDD or motherboard begin to get within 10% of their set upper limit .. SMFC mush take some action otherwise it will either thermal trip and go MAX or it will burn up... 


The way I currently tune by board is this..

for a given ambient temp... I determine how much fan I need while the processors are at 100% to say at least 10% below trip levels which I set manually for published TCaae .. that is the other thing here.. many supermicro boards are set up for worst case.. like E5-2699 whatever has the highest thermal load.. but if you run 55W CPUs in a BMC tuned for 160W .. its going to be unnecessarily noisy and waste a lot of watts on fan current it doesn't need.

Then I determine how much fan is required with the fans idle ... I like to see v4 chips idle at no more than 40c on a silent server .. 27-32 about normal in a noisier setup .. but 40s is fine .. as long as everything else on the board is happy as well.. that is the lower fan limit... in my current setup.. the onboard 4x 10g nics get way hotter than the CPUs so I have to have a bit more flow in the tunnel to keep them cool

and if you want to master class your setup.. you would instead of use a banded like SMFC.. you could spend a couple bucks and right size your fans

again .. supermicro or the integrator built the chassis with worse case in mind.. if you pull the stock 3.0A fans and let say in my case running 2x E5-2630L v4  .. which are specially binned processors that develop a lot less heat.. I put in .55A fans.. now I can let the BMC optimal curve eat .. because 50% (which is bmc target) fan curve on these fans is 1/4 the amp pull, RPM and CFM air flow than the original fans in the fan wall.

its about thermal dynamics.. you have so much heat to reject .. you need enough CFM to do the job.. and not an ounce more

so SMFC can either be smart enough to re-curve the proper fans to cool the proper components... or it cant... and if it cant.. something will burn up...

**Comment by duecedriver** [comment]
> > perhaps only using HDD temp
> 
> There aren't any FANA+, purely FAN0+ (edit: or did you mean use actual HDD temps in script, not from polling IPMI?). I'll just roll out a script. I have to account for MB_NIC_TEMP1 and MB_NIC_TEMP2 as well since they have no cooling (AOC-2UR6-i4XT on the other end of the motherboard, for example) and are done mostly by two of the fans from the far end just hoping to blow on them. I've been considering mounting a small 30x30x10mm fan directly onto their combined heat-sink and running a PWM cable to it off a spare port on a motherboard from one of the sections that usually has a lower fan speed, just to be safe about the power. I'm aware a tiny 3010 fan like that draws nearly nothing, I just don't willy-nilly draw from rails. Not as much air gets through if the PCIe slot is used, but especially if a rear-drive-kit is modded in (a rivet or two needed removal). Just trying to account for everything.
> 
> I wish I could describe the fan layout to the scripts such "these fans are behind these drives and affect this other sensor", that's why I'm thinking I'll roll my own given my needs.

For your board.. here is what you can do until SMFC becomes more configurable ... if ever...

either follow my suggestions above about replacing your 4 fans with lower CFM / Amp fans that will meet your cooling demand at 50% duty cycle which is what the current BMC is programmed to shoot for.. or take over manually

since you have IPMI compiled and running on a system somewhere.. it does not even need to be on the server in question and honestly easier and more convenient to run it remotely from a laptop etc... here is a small script you can modify or enter the commands individually 


The first line of setting the fans to maximum is mandatory.. and only has to be done once after the system boots... or you can set it from the IPMI/BMC web client on the host

the 'majic' happens on the other 2 lines...

the last 2 segments 0x00 0x05 are what slow the zones down

0x00 and 0x01 are the zones or fan groupings

the last 0xXX is a hex value for fan speed percentage .. again in hex ... so 0x64 would be 100%

ideally SMFC would query you for what sensor you want each of these 2 zones to key off and the min / max for that sensor... it would be easy but he is making this out to be nuclear physics... 

write a couple versions of the script if you wish.. silent at idle for example .. 0x10 (16%) is pretty quiet if your cpu and components stay cool enough.. good enough..

have another for when the room is hotter or the system is doing more work.. say 0x20) like 30%





#!/bin/sh

Echo Setting Fans Maximum
ipmitool -H 192.168.0.11 -U ADMIN -P PASSWORD -I lanplus raw 0x30 0x45 0x01 0x01

Sleep 5


Echo Setting Fans Custom

ipmitool -H 192.168.0.11 -U ADMIN -P PASSWORD -I lanplus raw 0x30 0x70 0x66 0x01 0x00 0x05

ipmitool -H 192.168.0.11 -U ADMIN -P PASSWORD -I lanplus raw 0x30 0x70 0x66 0x01 0x01 0x05




Echo Waiting 10 Seconds for Sensors

Sleep 10

ipmitool -H 192.168.0.11 -U ADMIN -P PASSWORD -I lanplus sdr | grep FAN
ipmitool -H 192.168.0.11 -U ADMIN -P PASSWORD -I lanplus sdr | grep CPU

**Comment by duecedriver** [comment]
> Hi [@MrMeeb](https://github.com/MrMeeb), I'm happy to investigate more on this. My assumption was that this motherboard has multiple CPU zones defined in IPMI. Some further questions:
> 
> * How many CPUs are installed on this board?
> * How many fans are installed on this board?
> * I assume you intentionally disabled HD zone in your configuration file, right?
> 
> Based on your report my assumption is that the CPU zone will be FAN1 and FAN2 and the HD zone will be FAN7 and FAN8. What if you run `ipmitool` with higher zone values like 2, 3, 4?

for this board.. designed for rack mount with a fan wall between hard drives at front.. and processors  lined up right to left.. 

zone 0x00 is for CPU0 on the right side of the board.. fans 1-4
zone 0x01 is for CPU1 on the left side of the motherboard fans 5-8

these are hot plug fans on a fan wall in most chassis configurations so you cant move the fan headers the fans are plugged into.. that has to remain as the mfg has it set

so your config.. if it bothered to ask this.. would allow zone one to fan curve off CPU0
and zone 0x01 to fan curve off CPU1

it would also keep watch either on secondary sensors the user could define.. 
or you could just parse the entire IPME sensor dump looking for any sensor getting close (10%) of their set max.. and then ramp all fans accordingly until values returned in limits and reverting back to primary sensor

**Comment by OdinVex** [comment]
> Hi [@duecedriver](https://github.com/duecedriver) and [@OdinVex](https://github.com/OdinVex), this is IPMI, please think about zones instead of individual fans. You can decide which zone will contain the most of the fans and how your cooling will work. This could be tough for server cases.

Thank you but no, thinking about zones in any of my servers cases (SYS-1028-TR4T+, 6028-TR4T+, etc) is the wrong way (for this case with my needs) to approach things because zones don't control individual fans in a way to target specific devices. Fans 7 and 8 are positioned behind the PCIe expansion slots and I use GPUs there, when I run some primitive AI-stuff I need them up higher on their own, nowhere else relating to those fans. When the MB_NIC temps rise (they're passively cooled in a pathetic way, bad design on Supermicro's part) I need 4 and 5 to spin up more. I always need fan 1 a little bit higher because of the PSU shrouds.

> My problem is that the configuration of `smfc` is already very complex and not really user-friendly. These steps would make it even more complex and abstract.

Now you know why I said I won't bother trying with SMFC but rather just stick to scripts. I've already carved it up with my own concepts. Just to point out for knowledge: There are no HDD temp fans in these cases. The fan "wall" in a SYS-6028-TR4T+ is all FAN0-8 (1-9), though only 4 of them are used. The fan "wall" in a SYS-1028-TR4T+ is all FAN0-8 (1-9) though only 8 are used. The "zones" concept just needs to take a backseat sometimes with manual definitions for targeting specific scenarios.

For now my script just sets each "zone" just to set "the most fans" to what they should be and then sets the target fans individually to what they need to be.

Thank you anyway, especially for the project.

Edit: For others with these cases the worst hot-spots are the MB_NICs (in my case 2 x Intel X540s on the AOC for 4x10GbE) and any GPUs (plus PSU 1 if you have a PSU split-shroud). Spinning up fans 7,8, (and an extra 9th fan if you want to attach a small fan within Supermicro's assumed power-rail's limits for the CPU1 side's fans) helps with PCIe-slot cards, 4 and 5 for MB NIC but you would probably be better off fitting a small 3010 fan directly to the shared AOC heatsink and running that manually or via 9th PWM connector).

**Comment by DeMoB** [comment]
@OdinVex, “Zones” are a Supermicro motherboard feature, not a smfc script concept.

Supermicro motherboards do not allow individual fan control, only setting an entire zone to one speed value, and there are only ever two zones on a Supermicro board.

So even in your script concepts, you’re still using Zones, because you’ll have a peripheral zone and a cpu zone, which if memory serves is even what Supermicro documentation refers to the two zones as.

**Comment by OdinVex** [comment]
> [@OdinVex](https://github.com/OdinVex), “Zones” are a Supermicro motherboard feature, not a smfc script concept.

That has nothing to do with what I've been talking about. Again, forget zones. Defining control over specific fans is what I've been talking about, repeatedly, *specifically*.

> Supermicro motherboards do not allow individual fan control, only setting an entire zone to one speed value, and there are only ever two zones on a Supermicro board.

Yes they do. I do it all the time, I've done it with three different X10-series systems (two have the same motherboard but different form-factors, 1U and 2U).

> So even in your script concepts, you’re still using Zones, because you’ll have a peripheral zone and a cpu zone, which if memory serves is even what Supermicro documentation refers to the two zones as.

Again, wrong, you can literally target any fan you want and set speeds.

**Comment by duecedriver** [comment]
> > [@OdinVex](https://github.com/OdinVex), “Zones” are a Supermicro motherboard feature, not a smfc script concept.
> 
> That has nothing to do with what I've been talking about. Again, forget zones. Defining control over specific fans is what I've been talking about, repeatedly, _specifically_.
> 
> > Supermicro motherboards do not allow individual fan control, only setting an entire zone to one speed value, and there are only ever two zones on a Supermicro board.
> 
> Yes they do. I do it all the time, I've done it with three different X10-series systems (two have the same motherboard but different form-factors, 1U and 2U).
> 
> > So even in your script concepts, you’re still using Zones, because you’ll have a peripheral zone and a cpu zone, which if memory serves is even what Supermicro documentation refers to the two zones as.
> 
> Again, wrong, you can literally target any fan you want and set speeds.

do tell .. give me the ipmi commands to target each fan on a x10DRU-i motherboard.. I am all ears.. 

supermicro programs these bios/bmc into zones in almost every case ...

personnely I dont think you know what you are talking about 

on otters MFG boards.. like intel boards (when they still made motherboards.  or higher end servers like Dell and HP .. you get better written an more feature rich BMC / SMC / FRU code... 

but in the case of supermicro.. its zones... 

so unless you are programming your own BMC software.. I have never seen individual fan control on a supermicro x10 series board

as far as power etc.. all the fans being in the same zone etc .. rails.. etc.. again lead me to believe you dont know what you are talking about when it comes to ATX standards, power supply design, etc...

**Comment by OdinVex** [comment]
> do tell .. give me the ipmi commands to target each fan on a x10DRU-i motherboard.. I am all ears..
> personnely I dont think you know what you are talking about

With the attitude you've had? No, I'll stick to my scripts.

> as far as power etc.. all the fans being in the same zone etc .. rails.. etc.. again lead me to believe you dont know what you are talking about when it comes to ATX standards, power supply design, etc...

As for rails, I mean the power delivery on the actual motherboards. You don't just hook up fans without knowing whether you'll draw too much current from the plane feeding however many it's designed to feed. Any basic electronics engineering course would help you to know that. You don't for instance throw a 1.5A completely-oversized fan on a Raspberry Pi 3 because it'll draw too much and they're limited to 2A, it'd starve the rest of the board and that's if it didn't overdraw and kill any cheaply designed circuits. Assumption isn't all you think it's cracked up to be. Unsubscribing, you're flat out toxic.

I will say this for the other guy: These X10s use an AST2400 for their BMC. The documentation's out there.

**Comment by duecedriver** [comment]
> > do tell .. give me the ipmi commands to target each fan on a x10DRU-i motherboard.. I am all ears..
> > personnely I dont think you know what you are talking about
> 
> With the attitude you've had? No, I'll stick to my scripts.
> 
> > as far as power etc.. all the fans being in the same zone etc .. rails.. etc.. again lead me to believe you dont know what you are talking about when it comes to ATX standards, power supply design, etc...
> 
> As for rails, I mean the power delivery on the actual motherboards. You don't just hook up fans without knowing whether you'll draw too much current from the plane feeding however many it's designed to feed. Any basic electronics engineering course would help you to know that. You don't for instance throw a 1.5A completely-oversized fan on a Raspberry Pi 3 because it'll draw too much and they're limited to 2A, it'd starve the rest of the board and that's if it didn't overdraw and kill any cheaply designed circuits. Assumption isn't all you think it's cracked up to be. Unsubscribing, you're flat out toxic.
> 
> I will say this for the other guy: These X10s use an AST2400 for their BMC. The documentation's out there.

no your just full of shit... 

As for rails, I mean the power delivery on the actual motherboards. You don't just hook up fans without knowing whether you'll draw too much current from the plane feeding however many it's designed to feed


the engineers at supermicro as well as every other server board mfg do all of this as basic power delivery design.. and have since the days where I was still loading code with punchcards.. 

if your scripts were so great.. post them up on GitHub.. or stop trolling around sites where folks are trying to help other folks out.. for the shortcomings of SMFC at least the author of this Github is trying to help folks out.. more than I can say for your sniveling ...

**Comment by petersulyok** [comment]
I converted this issue to the Discussion to give more room for new ideas...

**Comment by michaels2408** [comment]
I also am using the X10DRU-i+ board in a 2u case where 4 fans are run. Not sure why the previous person was trying so hard to drill down to control each individual fan, but this board does in fact have two zones, 0 and 1. Zone 0 controls fans 1-4, zone 1 controls fans 5-8. Difference is there is no Hard Drive Zones. Zone 0; fans 1-4;  cool 1/2 the hard drive bays, CPU 1 with its associated memory banks, and any peripherals downstream. Zone 1 does the exact same for CPU 2. So in essence your script would just need to treat the HD_ZONE as CPU 2. This would solve the problem of 1 cpu always getting too hot. I am making an attempt to modify the scripts myself, but as you have already eluded to the programming is complex and it will take time to trace through to ensure correcting the zone issue does not break something else.

**Reply by petersulyok** [reply]
Hi @michaels2408, in the current v4.0.0b5 release you can define freely the IPMI zone id to your zones in the configuration file. I'm also thinking on making this feature more general and one heat source (e.g. CPU) could control several IPMI zones.

**Reply by petersulyok** [reply]
In `v4.0.0b6` pre-release multiple IPMI zones can be assigned to a fan controller (heat source), as well.
Also, see [my comment here](https://github.com/petersulyok/smfc/discussions/55#discussioncomment-13072987).
